protocol/src/main/java/net/md_5/bungee/protocol/channel/BungeeChannelInitializer.java

```diff
@@ -0,0 +1,84 @@
+package net.md_5.bungee.protocol.channel;
+
+import com.google.common.base.Preconditions;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelInitializer;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+/*
+ * This class holds all minecraft related channel initializers. BungeeCord will set these values
+ * on startup, and they can be used by third party plugins to modify the channel pipeline.
+ *
+ * Please note that this API is unsafe and doesn't provide any guarantees about the stability of the
+ * channel pipeline. Use at your own risk.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public abstract class BungeeChannelInitializer
+{
+    /**
+     * Holds the channel initializer for incoming connections
+     */
+    @Getter
+    private static BungeeChannelInitializer frontendHolder;
```

Comment:
I think these shouldn't be static

Reply:
Should i put the non static objects into BungeeCord.java and make them accessable via 

ProxyServer.getInstance().unsafe()?